vllm/model_executor/layers/quantization/utils/fp8_utils.py

```diff
@@ -240,87 +251,83 @@ def _run_deepgemm(
         self,
         input_2d: torch.Tensor,
         weight: torch.Tensor,
-        block_size: list[int],
         weight_scale: torch.Tensor,
     ) -> torch.Tensor:
         # ensure DeepGEMM-backed custom op is registered before use
         import vllm.model_executor.layers.quantization.deepgemm  # noqa: F401
 
-        q_input, x_scale = per_token_group_quant_fp8(
-            input_2d,
-            block_size[1],
-            column_major_scales=True,
-            use_ue8m0=self.ue8m0_deepgemm_supported,
-        )
+        assert self.deepgemm_input_quant_op is not None
+        q_input, x_scale = self.deepgemm_input_quant_op.forward_cuda(input_2d)
         return torch.ops.vllm.w8a8_block_fp8_matmul_deepgemm(
             q_input,
             weight,
             x_scale,
             weight_scale,
-            block_size,
+            self.weight_group_shape,
             output_dtype=input_2d.dtype)
 
     def _run_cutlass(
         self,
         input_2d: torch.Tensor,
         weight: torch.Tensor,
-        block_size: list[int],
         weight_scale: torch.Tensor,
     ) -> torch.Tensor:
-        q_input, x_scale = per_token_group_quant_fp8(input_2d,
-                                                     block_size[1],
-                                                     column_major_scales=True,
-                                                     use_ue8m0=False)
+        q_input, x_scale = self.input_quant_op.forward_cuda(input_2d)
```

Comment:
If you call `forward_cuda` directly we won't ever dispatch to torch - just call the op directly and add `compilation_config.custom_ops.append( ["+quant_fp8"])` in config for non-Blackwell arches.

Reply:
Unless there's another reason we want to always dispatch to the CUDA impl here (then we should resolve that)